### PR TITLE
Fix nil pointer panic in kvstore

### LINKED ISSUE #2001: kvstore panics on empty map
## Expected Behavior
Setting a key on an empty kvstore, i.e. ConfigMap with no data field, should not panic.

## Actual Behavior

```bash
panic: assignment to entry in nil map [recovered]
	panic: assignment to entry in nil map
```

## Steps to Reproduce the Problem

1. Create empty ConfigMap (name/namespace only)
2. Use kvstore
3. Set key -> panic

### PR DESCRIPTION
**Pull request by embano1**
Signed-off-by: Michael Gasch <mgasch@vmware.com>

# Changes

- Fix bug in kvstore causing nil pointer panic on an empty map (map without data)

/kind  bug

Fixes #2001 

**Release Note**

```release-note

```

**Docs**

```docs

```

### CODE REVIEW

File: kvstore/kvstore_cm_test.go [hunk 1]
```diff
@@ -139,6 +139,49 @@ func TestLoadSaveUpdate(t *testing.T) {
...
+
+	if tc.updated == nil {
+		t.Errorf("ConfigMap Not updated")
+	}
+	var ret string
+	err := cs.Get(context.Background(), "jimmy", &ret)
+	if err != nil {
+		t.Error("failed to return string:", err)
+	}
+	if err != nil {
```
Comment (vaikas) [inline kvstore/kvstore_cm_test.go:177]: duplicate err check here?

**Comment by vaikas** [comment]
Downstream tests are unrelated to this change and being worked on in downstreams.

File: kvstore/kvstore_cm_test.go (hunk 1 above)
Comment (embano1) [reply kvstore/kvstore_cm_test.go]: Copy pasta from other tests :)

**Comment by vaikas** [comment]
Thanks!
/lgtm
/approve

**Comment by dprotaso** [comment]
/retest

metric flakes

**Pull request merged**